### PR TITLE
Llama2 node-llama-cpp chat module

### PR DESCRIPTION
**Pull request by nigel-daniels**
This is a chat module based on `node-llama-cpp` it will allow you to use basic chat module functionality with Llama2 or any modules supported by node-llama-cpp and ultimately llama.cpp.

### CODE REVIEW

File: langchain/package.json [hunk 1]
```diff
@@ -758,7 +761,7 @@
     "mongodb": "^5.2.0",
```
[the remark this answers is not in the document]
Comment (nigel-daniels) [reply langchain/package.json:761]: FYI I tested against LLM version and there were no regressions.

**Comment by nigel-daniels** [comment]
Currently this module supports call but not stream, I was also not sure if there were other methods that require implementation. Please advise if more is needed or if this should be added to the docs, etc. Also during testing I could not get the memory chain to give good results but this may be a limitation of my test env. (I can only support Llama2-13G). Using Llama2-7G other tests gave strange/bad results so I think it is a limitation of the LLM.

**Comment by jacoblee93** [comment]
Phenomenal, thank you!

**Comment by jacoblee93** [comment]
> Currently this module supports `call` but not `stream`, I was also not sure if there were other methods that require implementation. Please advise if more is needed or if this should be added to the docs, etc. Also during testing I could not get the memory chain to give good results but this may be a limitation of my test env. (I can only support Llama2-13G). Using Llama2-7G other tests gave strange/bad results so I think it is a limitation of the LLM.

No problem, `.stream()` will just fall back to `._call`.

**Comment by nigel-daniels** [comment]
Great, in that case I'll look at getting an embedding done to go with this, as work allows :)

**Pull request merged**